drivers/sensor/lm75/lm75.c

```diff
@@ -64,13 +64,13 @@ static int lm75_sample_fetch(const struct device *dev,
 {
 	struct lm75_data *data = dev->data;
 	const struct lm75_config *cfg = dev->config;
-	enum pm_device_state pm_state;
 	int ret;
+	int pm_ret;
 
-	(void)pm_device_state_get(dev, &pm_state);
-	if (pm_state != PM_DEVICE_STATE_ACTIVE) {
-		ret = -EIO;
-		return ret;
+	pm_ret = pm_device_runtime_get(dev);
+	if (pm_ret < 0) {
+		LOG_ERR("pm_device_runtime_get failed with error %i", pm_ret);
+		return pm_ret;
 	}
```

Comment:
there's no need for `pm_ret`, just use `ret`

Reply:
@gmarull see the conversation with @aurel32, If I use ret, I either cant log the PM error or cant return the fetch error.

Reply:
you should implement a single exit point that puts the device in case of failure once it has been obtained.

Reply:
ok, but I still dont see how thats possible with only one Variable, but maybe im just blind :D btw, I mentioned you on discord/sensors

Reply:
Thanks for your thoughts on this, I think this is a good idea!
I also noticed that these errors are often not handled, but I wanted to do it better because the lack of checking Return codes in zephyr has caused me longer-than-needed debugging sessions more than once.
That being said, I also like the Idea of printing the error in the handler more than my current approach, especially because this will mean that all code that uses the function will now have logging.
If everyones fine with it, I can add the logging with this PR too.

Reply:
> If everyones fine with it, I can add the logging with this PR too.

I'd personally just have a separate PR for that, given it falls a bit off-topic, but no strong objection.